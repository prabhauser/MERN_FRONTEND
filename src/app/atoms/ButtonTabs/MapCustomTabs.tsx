import ButtonTabs from './index';

type Props = {
    navigateTab: Function;
    tabsArray: string[];
    currTab: Number;
    setCurrTab: Function;
    className?: string;
    position?: string;
};

const MapCustomTabs = (props: Props) => {
    const { tabsArray, currTab, setCurrTab, navigateTab, className, position } = props;
    return (
        <div className="pad-cont-10">
            <ButtonTabs
                tabsArray={tabsArray}
                selectedTab={currTab}
                setSelectedTab={setCurrTab}
                className={className}
                position={position}
            />
            {navigateTab()}
        </div>
    );
};

export default MapCustomTabs;
