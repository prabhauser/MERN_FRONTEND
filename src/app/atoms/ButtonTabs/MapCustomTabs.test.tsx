import { shallow } from 'enzyme';
import MapCustomTabs from './MapCustomTabs';

describe('MapCustomTabs', () => {
    describe('MapCustomTabs snapshot', () => {
        const tabsArray = [
            'Supervisors',
            'Municipal Links(ML)',
            'SWA',
            'Health Facilities',
            'Educational Facilities',
            'Households'
        ];
        it('should take customtab snapshot', () => {
            const component = shallow(
                <MapCustomTabs
                    tabsArray={tabsArray}
                    navigateTab={() => void 0}
                    currTab={0}
                    setCurrTab={() => {}}
                    className={'test'}
                    position={'relative'}
                />
            );
            expect(component).toMatchSnapshot();
        });
    });
});
