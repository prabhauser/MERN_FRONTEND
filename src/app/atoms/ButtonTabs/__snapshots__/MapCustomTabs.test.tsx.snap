// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapCustomTabs MapCustomTabs snapshot should take customtab snapshot 1`] = `
<div
  className="pad-cont-10"
>
  <ButtonTabs
    className="test"
    position="relative"
    selectedTab={0}
    setSelectedTab={[Function]}
    tabsArray={
      Array [
        "Supervisors",
        "Municipal Links(ML)",
        "SWA",
        "Health Facilities",
        "Educational Facilities",
        "Households",
      ]
    }
  />
</div>
`;
